### PR TITLE
add Prefab registration and Library Import

### PR DESCRIPTION
**Pull request by aquisenberry**

### CODE REVIEW

**Comment by ericlathrop** [comment]
FYI I might not get a chance to merge this in until I get back from GDC. I'm going to make a few small nitpick comments, that either you or I can take care of.

File: lib/game.js [hunk 1]
```diff
@@ -171,5 +173,15 @@ Game.prototype.instantiatePrefab = function(name) {
 	}.bind(this));
 	return id;
 };
-
-module.exports = Game;
+Game.prototype.registerPrefab = function (name,obj){
+	var prefabs = this.prefabs;
+	prefabs[name] = obj;
+};
+Game.prototype.importPrefabs = function(library){
```
Comment (ericlathrop) [inline lib/game.js]: This should just be called `registerPrefabs` and the argument should be named `prefabs`

File: lib/game.js [hunk 2]
```diff
@@ -171,5 +173,15 @@ Game.prototype.instantiatePrefab = function(name) {
 	}.bind(this));
 	return id;
 };
-
-module.exports = Game;
+Game.prototype.registerPrefab = function (name,obj){
+	var prefabs = this.prefabs;
```
Comment (ericlathrop) [inline lib/game.js]: You don't really need this variable.

File: lib/game.js [hunk 3]
```diff
@@ -171,5 +173,15 @@ Game.prototype.instantiatePrefab = function(name) {
...
+	prefabs[name] = obj;
```
Comment (ericlathrop) [inline lib/game.js]: Just make this line `this.prefabs[name] = obj;`

File: lib/game.js [hunk 4]
```diff
@@ -171,5 +173,15 @@ Game.prototype.instantiatePrefab = function(name) {
 	}.bind(this));
 	return id;
 };
-
-module.exports = Game;
+Game.prototype.registerPrefab = function (name,obj){
```
Comment (ericlathrop) [inline lib/game.js]: `obj` should be named `components`.

File: lib/game.js [hunk 5]
```diff
@@ -171,5 +173,15 @@ Game.prototype.instantiatePrefab = function(name) {
...
+	var prefabs = this.prefabs;
+	prefabs[name] = obj;
+};
+Game.prototype.importPrefabs = function(library){
+	var game = this;
```
Comment (ericlathrop) [inline lib/game.js]: You don't need this variable. Just add a `.bind(this)` to the end of the anonymous function so `this.registerPrefab()` works inside of it.

File: lib/game.js (hunk 1 above)
Comment (aquisenberry) [inline lib/game.js]: Should I only expose `registerPrefabs` to the user then?

Comment (ericlathrop) [inline lib/game.js]: Do you mean hide `registerPrefab`? No, I think both of them are useful. If you think there should only be one method, I'd get rid of `importPrefabs`/`registerPrefabs` and let the user do the loop.

**Comment by ericlathrop** [comment]
Thanks! :heart:

**Pull request merged**